commit ecfad20871e855d9e54d3a528cf9219ce5c93626
Author: Bastien Nocera <hadess@hadess.net>
Date:   Mon Nov 1 16:29:09 2010 +0000

    keymap: Add force release for HP touchpad off
    
    Force the touchpad off/on keys getting released, as they usually
    only send a "repeat".
    
    https://bugzilla.redhat.com/show_bug.cgi?id=623239
    
    Signed-off-by: Martin Pitt <martin.pitt@ubuntu.com>

diff --git a/Makefile.am b/Makefile.am
index 3a95046..032eb28 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -601,6 +601,7 @@ dist_udevkeymap_DATA = \
 
 udevkeymapforcereldir = $(libexecdir)/keymaps/force-release
 dist_udevkeymapforcerel_DATA = \
+	extras/keymap/force-release-maps/hp-other \
 	extras/keymap/force-release-maps/samsung-other \
 	extras/keymap/force-release-maps/common-volume-keys
 
diff --git a/extras/keymap/95-keyboard-force-release.rules b/extras/keymap/95-keyboard-force-release.rules
index 562dc8d..6f7c0fd 100644
--- a/extras/keymap/95-keyboard-force-release.rules
+++ b/extras/keymap/95-keyboard-force-release.rules
@@ -35,4 +35,9 @@ ENV{DMI_VENDOR}=="TOSHIBA", ATTR{[dmi/id]product_name}=="Satellite U300|Satellit
 
 ENV{DMI_VENDOR}=="Viooo Corporation", ATTR{[dmi/id]product_name}=="PT17", RUN+="keyboard-force-release.sh $devpath common-volume-keys"
 
+# These are all the HP laptops that setup a touchpad toggle key
+ENV{DMI_VENDOR}=="Hewlett-Packard*", ATTR{[dmi/id]product_name}=="*[pP][aA][vV][iI][lL][iI][oO][nN]*", RUN+="keyboard-force-release.sh $devpath hp-other"
+ENV{DMI_VENDOR}=="Hewlett-Packard*", ATTR{[dmi/id]product_name}=="*[tT][xX]2*", RUN+="keyboard-force-release.sh $devpath hp-other"
+ENV{DMI_VENDOR}=="Hewlett-Packard*", ATTR{[dmi/id]product_name}=="*2510p*|*2530p*|HP G60 Notebook PC", RUN+="keyboard-force-release.sh $devpath hp-other"
+
 LABEL="force_release_end"
diff --git a/extras/keymap/force-release-maps/hp-other b/extras/keymap/force-release-maps/hp-other
new file mode 100644
index 0000000..6621370
--- /dev/null
+++ b/extras/keymap/force-release-maps/hp-other
@@ -0,0 +1,3 @@
+# list of scancodes (hex or decimal), optional comment
+0xd8 # Touchpad off
+0xd9 # Touchpad on
